doc/howtos/rdtraining/sprinkles.rst

```diff
@@ -0,0 +1,481 @@
+
+    - Add the following field:
+
+    =================================== ======================= =======================
+    Model                               Field                   Type
+    =================================== ======================= =======================
+    ``estate.property.type``            Sequence                Integer
+    =================================== ======================= =======================
+
+    - Add the sequence to the ``estate.property.type`` list view with the right widget.
+
+    Tip: you can find an example here:
+    `model <https://github.com/odoo/odoo/blob/892dd6860733c46caf379fd36f57219082331b66/addons/crm/models/crm_stage.py#L36>`__
+    and
+    `view <https://github.com/odoo/odoo/blob/892dd6860733c46caf379fd36f57219082331b66/addons/crm/views/crm_stage_views.xml#L23>`__.
+
+Attributes and options
+======================
+
+It would be prohibitive to detail all the available features which allows fine tuning the look of a
+view. Therefore, we'll only stick to the most common ones.
+
+Form
+----
+
+.. note::
+
+    **Goal**: at the end of this section, the property form view has:
+    - Conditional display of buttons and fields
+    - Tag colors
+
+    .. image:: sprinkles/media/form.gif
+      :align: center
+      :alt: Form view with sprinkles
+
+In our real estate module, we want to modify the behavior of some fields. For example, we don't
+want to be able to create or edit a property type from the form view. Indeed, we expect the
+types to be handled in their appropriate menu. Moreover, we want to give tag colors. Several field
+widgets take the ``options`` attribute in to customize their behavior.
+
+.. exercise:: Widget options
+
+    - Add the appropriate option to the ``property_id`` field to prevent the creation and the
+      edition of a property type from the property form view. Have a look at the
+      :ref:`Many2one widget documentation <reference/js/widgets>` for more info.
+
+    - Add the following field:
+
+    =================================== ======================= =======================
+    Model                               Field                   Type
+    =================================== ======================= =======================
+    ``estate.property.tag``             Color                   Integer
+    =================================== ======================= =======================
+
+    Then, add the appropriate option to the ``tag_ids`` field to add a color picker on the tags.
+    Have a look at the :ref:`FieldMany2ManyTags widget documentation <reference/js/widgets>`
+    for more info.
+
+In the :ref:`howto/rdtraining/firstui` chapter, we saw that reserved fields was used for
+specific behaviors. For example, the ``active`` field is used to automatically filter out
+inactive records. We added the ``state`` as a reserved field as well. It's not time to use it!
+
+.. exercise:: Conditional display of buttons
```

Comment:
Maybe this exercise must be fusion with the next one. Here we ask for display buttons conditionally, but we only talk about invisible in buttons in the next paragraph. Also, to make this exercise you need attrs that is explained after.

Maybe explain the conditions as is done in the next exercise, it's not very intuitive from the gif.

Reply:
You can do the exercise with the `states=` attribute.